### PR TITLE
Cart wishlist

### PR DESCRIPTION
**Pull request by suryxks**
features implemented:

- add to cart
- increase and decrease product quantity in cart
- delete from cart
- move item from cart to wish list
- add to wish list 
- remove from wish list 
username: `adarshbalika@gmail.com`
password: `adarshBalika123`

### CODE REVIEW

File: src/Pages/Cart/Cart.jsx [hunk 1]
```diff
@@ -1,13 +1,58 @@
-import React from 'react'
-import {Navbar} from '../../components'
+import React, { useEffect } from 'react'
+import { Navbar } from '../../components'
+import { useCart } from '../../contexts/CartContext'
+import CartCard from '../../components/HorizontalCard/CartCard'
+import { useWishList } from '../../contexts/WishListContext';
+import toast, { Toaster } from 'react-hot-toast';
```
Comment (MiheerTamkhane) [inline src/Pages/Cart/Cart.jsx:6]: Use barrel exports, code will more clean

File: src/Pages/Wishlist/WishList.jsx [hunk 2]
```diff
@@ -1,11 +1,48 @@
...
+                const {
+                    _id:id,
+                    title,
+                    description,
+                    price,
+                    discount,
+                    imageUrl,
+                    rating,
+                    Originalprice
+                  } = product;
```
Comment (MiheerTamkhane) [inline src/Pages/Wishlist/WishList.jsx:27]: This is good to destructure the product here code looks neat and clean

File: src/components/Navbar/Navbar.jsx [hunk 3]
```diff
@@ -32,9 +36,12 @@ const Navbar = () => {
           </button>
           <Link to="/wishlist">
             <FavoriteBorderIcon className="icon" />
+            {wishList.length!=0?(<span className='badge-icon'>{wishList.length===0?null:wishList.length}</span>):null}
```
Comment (MiheerTamkhane) [inline src/components/Navbar/Navbar.jsx:39]: you can also use and use double check like this "!=="
wishlist.length!==0 && <span> 
and 
within span same , wishlist.length > 0 && wishlist.length

File: src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -32,9 +36,12 @@ const Navbar = () => {
...
           </Link>
           <Link to="/cart">
             <ShoppingBagIcon className="icon" />
+           {cart.length!=0?(<span className='badge-icon'>{cart.length===0?null:cart.length}</span>):null}
```
Comment (MiheerTamkhane) [inline src/components/Navbar/Navbar.jsx:43]: Same here

**Review by MiheerTamkhane — APPROVED** [review]
Make changes i have commented, Other than that code is readable and clean
Great work

**Review by manojkumar8400 — APPROVED** [review]
good to go.

**Pull request merged**